bindings/java/src/main/native/Makefile.am

```diff
@@ -11,34 +11,34 @@ javadir=$(top_srcdir)/bindings/java
 MVNCMD=$(MVN) -B -T 1C -f $(topdir)/bindings/java/pom.xml -Dmaven.repo.local=$(java_build_dir)/.deps \
               -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
 
-BUILT_SOURCES = org_openucx_jucx_ucp_UcpConstants.h \
-                org_openucx_jucx_ucp_UcpContext.h \
-                org_openucx_jucx_ucp_UcpEndpoint.h \
-                org_openucx_jucx_ucp_UcpWorker.h \
-                org_openucx_jucx_ucs_UcsConstants.h \
-                org_openucx_jucx_ucp_UcpListener.h
-
-DISTCLEANFILES = org_openucx_jucx_ucp_UcpConstants.h \
-                 org_openucx_jucx_ucp_UcpContext.h \
-                 org_openucx_jucx_ucp_UcpEndpoint.h \
-                 org_openucx_jucx_ucp_UcpWorker.h \
-                 org_openucx_jucx_ucs_UcsConstants.h \
-                 org_openucx_jucx_ucp_UcpListener.h
-
-org_openucx_jucx_ucp_UcpListener.h:
-org_openucx_jucx_ucp_UcpConstants.h:
-org_openucx_jucx_ucp_UcpEndpoint.h:
-org_openucx_jucx_ucp_UcpWorker.h:
-org_openucx_jucx_ucs_UcsConstants.h:
-org_openucx_jucx_ucp_UcpContext.h:
-	$(MVNCMD) compile native:javah
+JUCX_GENERATED_H_FILES = org_openucx_jucx_ucp_UcpConstants.h \
+                         org_openucx_jucx_ucp_UcpContext.h \
+                         org_openucx_jucx_ucp_UcpEndpoint.h \
+                         org_openucx_jucx_ucp_UcpWorker.h \
+                         org_openucx_jucx_ucs_UcsConstants.h \
+                         org_openucx_jucx_ucp_UcpListener.h \
+                         org_openucx_jucx_ucp_UcpMemory.h \
+                         org_openucx_jucx_ucp_UcpRemoteKey.h
+
+BUILT_SOURCES = $(JUCX_GENERATED_H_FILES)
+
+DISTCLEANFILES = $(JUCX_GENERATED_H_FILES)
+
+MOSTLYCLEANFILES = $(JUCX_GENERATED_H_FILES)
+
+org_openucx_jucx_ucp_UcpContext.h: $(javadir)/src/main/java/org/openucx/jucx/ucs/*.java \
+                                   $(javadir)/src/main/java/org/openucx/jucx/ucp/*.java
+        $(MVNCMD) compile native:javah
+
+$(JUCX_GENERATED_H_FILES):
```

Comment:
missing compile command?

Reply:
The issue that if put here:
`$(JUCX_GENERATED_H_FILES)` then if you'll change java file it'll generate headers 10 times for each header.